### ISSUE TITLE
[BUG] Panic when calling .bytes() on R2 ObjectBody representing range of very large object 

### ISSUE THREAD

**Issue by DerZade** [cloudflare/workers-rs#269]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What version of `workers-rs` are you using?

main branch

### Describe the bug

There is a `panicked at 'capacity overflow'` error, when calling `object_body.bytes()` on an object with a size of multiple GBs, even if there was only a very small range requested.

### Steps To Reproduce

```rs
let obj = bucket
             .get("my_file_with_multiple_gb")
             .range(Range::OffsetWithLength { offset: 0, length: 127 })
             .execute()
             .await
             .unwrap()
             .unwrap();

let obj_body =  obj.body().unwrap();

let bytes = obj_body.bytes().await.unwrap(); // this panics
```

### How to fix

The following line should probably take the length of the range (if there is any) and only use the size as a fallback:

https://github.com/cloudflare/workers-rs/blob/b55bb592ee6dbf11fa1971f2c93055ab121393b6/worker/src/r2/mod.rs#L253

**Comment by DerZade** [comment]
Maybe #277 resolves this.

**Comment by DerZade** [comment]
I can confirm that this is indeed fixed by #277.

**Comment by DerZade** [comment]
I'll close this, since #277 was merged 🙃
